### PR TITLE
Move to Cargo workspaces

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14381)

### CODE REVIEW

**Comment by nox** [comment]
Cc @larsbergstrom

File: python/servo/devenv_commands.py [hunk 1]
```diff
@@ -89,11 +89,9 @@ def update_cargo(self, params=None, package=None, all_packages=None):
 
         self.ensure_bootstrapped()
 
-        for cargo_path in CARGO_PATHS:
```
Comment (SimonSapin) [inline python/servo/devenv_commands.py]: r=me with `CARGO_PATHS` removed near the top of the file.

**Comment by SimonSapin** [comment]
@larsbergstrom in https://github.com/servo/servo/pull/12391#issuecomment-245337056:

> Ensure that `cargo vendor` on `ports/geckolib` does NOT pull in the full dependency graph

TL;DR: We’re good. It looks like, when a path dependency is in a workspace, that workspace is completely ignored.

----

mozilla-central has a single `gkrust` crate in its `toolkit/library/rust` directory that (recursively) depends on all the Rust code included in Gecko. In the `incubator/stylo` repository, there’s (indirectly) a path dependency to `../../../servo/ports/geckolib`.

To get a similar but simplified setup, I created a temporary directory with this `Cargo.toml`:

```toml
[package]
name = "vendor-test"
version = "0.0.0"

[dependencies]
geckoservo = {path = "/home/simon/projects/servo2/ports/geckolib"}
```

… where `servo2` has this PR checked out. Then I ran `cargo update`, `cargo install cargo-vendor`, and `cargo vendor`. This creates a `Cargo.lock` file that contains the line `[package]` 61 times, and a `vendor` directory that contains 55 sub-directories. (The difference is the number of recursive path dependencies.)

For comparison, `/home/simon/projects/servo2/Cargo.lock` contains the line `[package]` exactly 300 times.

**Pull request merged**